MoreLinq/ScanRight.cs

```diff
@@ -0,0 +1,105 @@
+﻿#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2017 Leandro F. Vieira (leandromoh). All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Peforms a right-associative scan (inclusive prefix) on a sequence of elements.
+        /// This operator is the right-associative version of the 
+        /// <see cref="MoreEnumerable.Scan{TSource}(IEnumerable{TSource}, Func{TSource, TSource, TSource})"/> LINQ operator.
+        /// </summary>
+        /// <typeparam name="TSource">Type of elements in source sequence.</typeparam>
+        /// <param name="source">Source sequence.</param>
+        /// <param name="func">A right-associative accumulator function to be invoked on each element.</param>
```

Comment:
I think it would be useful to describe here the order of the arguments (first argument is the current value in the sequence, second argument is the previous accumulator value), because the types are not giving us any hints here.